lua/lvim/core/cmp.lua

```diff
@@ -156,62 +156,66 @@ M.config = function()
       fields = { "kind", "abbr", "menu" },
       max_width = 0,
       kind_icons = lvim.icons.kind,
-      source_names = {
-        nvim_lsp = "(LSP)",
-        emoji = "(Emoji)",
-        path = "(Path)",
-        calc = "(Calc)",
-        cmp_tabnine = "(Tabnine)",
-        vsnip = "(Snippet)",
-        luasnip = "(Snippet)",
-        buffer = "(Buffer)",
-        tmux = "(TMUX)",
-        copilot = "(Copilot)",
-        treesitter = "(TreeSitter)",
-      },
-      duplicates = {
-        buffer = 1,
-        path = 1,
-        nvim_lsp = 0,
-        luasnip = 1,
+      source_settings = {
+        copilot = {
+          menu_name = "(Copilot)",
+          icon = lvim.icons.git.Octoface,
+          hl_group = "CmpItemKindCopilot",
+        },
+        emoji = {
+          menu_name = "(Emoji)",
+          icon = lvim.icons.misc.Smiley,
+          hl_group = "CmpItemKindEmoji",
+        },
+        cmp_tabnine = {
+          menu_name = "(Tabnine)",
+          icon = lvim.icons.misc.Robot,
+          hl_group = "CmpItemKindTabnine",
+        },
+        crates = {
+          menu_name = "(Crates)",
+          icon = lvim.icons.misc.Package,
+          hl_group = "CmpItemKindCrate",
+        },
+        ["lab.quick_data"] = {
+          menu_name = "(QuickData)",
+          icon = lvim.icons.misc.CircuitBoard,
+          hl_group = "CmpItemKindConstant",
+        },
+        nvim_lsp = {
+          menu_name = "(LSP)",
+          duplicates = 0,
+        },
+        path = {
+          menu_name = "(Path)",
+          duplicates = 1,
+        },
+        luasnip = {
+          menu_name = "(Snippet)",
+          duplicates = 1,
+        },
+        buffer = {
+          menu_name = "(Buffer)",
+          duplicates = 1,
+        },
+        calc = { menu_name = "(Calc)" },
+        vsnip = { menu_name = "(Snippet)" },
+        treesitter = { menu_name = "(TreeSitter)" },
+        tmux = { menu_name = "(TMUX)" },
       },
       duplicates_default = 0,
       format = function(entry, vim_item)
         local max_width = lvim.builtin.cmp.formatting.max_width
+        local source_settings = lvim.builtin.cmp.formatting.source_settings[entry.source.name]
```

Comment:
you need to make sure this exists

```suggestion
        local source_settings = vim.tbl_get(lvim.builtin.cmp.formatting.source_settings, entry.source.name) or {}
```